Services/Administration/OJS.Services.Administration.Business/Validation/Factories/Implementations/ProblemResourceValidatorsFactory.cs

```diff
@@ -0,0 +1,96 @@
+namespace OJS.Services.Administration.Business.Validation.Factories.Implementations;
+
+using AutoCrudAdmin.Models;
+using AutoCrudAdmin.ViewModels;
+using OJS.Common.Enumerations;
+using OJS.Data.Models.Problems;
+using OJS.Services.Administration.Business.Validation.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GeneralResource = OJS.Common.Resources.AdministrationGeneral;
+using Resource = OJS.Common.Resources.ProblemResourcesController;
+
+public class ProblemResourceValidatorsFactory : IProblemResourceValidatorsFactory
+{
+    private readonly IProblemsValidationHelper problemsValidationHelper;
```

Comment:
Validations helpers should not be called inside a validator.

Validators should get the data to validate as arguments.

**Validators should not make any db calls in any way, i.e. thought another class**

Reply:
Why Validations helpers should not be called inside a validator? How to reuse validation logic if so?
The other 2 I agree.

Reply:
In the case with the Validators Factories, the parameters are not very flexible, as they come directly from the auto crud admin and it is quite difficult to pass whatever you want, hence the db calls..

Reply:
> Why Validations helpers should not be called inside a validator? How to reuse validation logic if so? The other 2 I agree.

Validator helpers with DB calls.

Reply:
> In the case with the Validators Factories, the parameters are not very flexible, as they come directly from the auto crud admin and it is quite difficult to pass whatever you want, hence the db calls..

This can be circumvented with `params` args of type `object`, i.e. `params object[] args`.

It will be uglier, but cleaner.

Reply:
I made separate issue for this: https://github.com/SoftUni-Internal/exam-systems-issues/issues/122